client/src/component/Search.js

```diff
@@ -1,45 +1,62 @@
 import "./User.css";
-import { useContext } from "react";
+import { useContext,useState } from "react";
 import { AppContext } from "../AppContext";
+import { useNavigate } from "react-router-dom";
 
 export const Search = () => {
-    const { userName, setCodewarsData, setGitHubData } = useContext(AppContext);
-    const handleSearch = () => {
-        /////****GETTING CODEWARS DATA****///////
-        fetch(`https://www.codewars.com/api/v1/users/${userName}`)
-            .then((res) => {
-                if (!res.ok) {
-                    throw new Error(res.statusText);
-                }
-                return res.json();
-            })
-            .then((data) => {
-                return setCodewarsData(data);
-            })
-            .catch((err) => {
-                console.error(err);
-            });
-        ///////GETING GITHUB PULL REQUEST ON CYF REPOSITORI
-        fetch(`https://api.github.com/search/issues?q=is:pr%20author:${userName}%20user:codeyourfuture`)
-            .then((res) => {
-                if (!res.ok) {
-                    throw new Error(res.statusText);
-                }
-                return res.json();
-            })
-            .then((data) => {
-                return setGitHubData(data);
-            })
-            .catch((err) => {
-                console.error(err);
-            });
-    };
-    return (
-        <button className="user-button" onClick={() => handleSearch()} >search</button>
-    );
-};
-
+	const { userName, setCodewarsData, setGitHubData } = useContext(AppContext);
+    const navigate = useNavigate();
+    const [error, setError] = useState(false);
 
+	const handleSearch = () => {
+		/////****GETTING CODEWARS DATA****///////
+		fetch(`https://www.codewars.com/api/v1/users/${userName}`)
+			.then((res) => {
+				if (!res.ok) {
+					throw new Error(res.statusText);
+				}
+				return res.json();
+			})
+			.then((data) => {
+                navigate("/result");
+				return setCodewarsData(data);
 
+			})
+			.catch((err) => {
+				console.error(err);
+                setError(true);
+                });
+                setTimeout(() => {
+                    setError(false);
+                  }, 3000);
```

Comment:
Maybe you should leave the error there until the user starts to type in the field rather than removing after 3 seconds as it will be invalid until there is a change so we dont want to remove it